### DISCUSSION TITLE
Error message under SD.NEXT Version 9bf08c5

### DISCUSSION THREAD

**Discussion by ThePixelDiffusionPirate** [wkpark/sd-webui-model-mixer#114] [Q&A] [answered]
Hello, I don't know if this is a bug or if Model-Mixer works at all under SD.Next, so I'm asking here first. Does anyone have SD.Next running under Win11 with Model-Mixer?

I have quite a few things disabled because I have an AMD graphics card, could it be related to that or rather the SD.Next structure?

Error

12:55:58-788895 ERROR    Module load: extensions\sd-webui-model-mixer\scripts\model_mixer.py: ImportError

╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮

│ C:\Users\TestBuild\Diffusion\automatic\modules\script_loading.py:25 in load_module                    │

│                                                                                                                      │

│   24 │   │   │   with contextlib.redirect_stdout(io.StringIO()) as stdout:                                           │

│ ❱ 25 │   │   │   │   module_spec.loader.exec_module(module)                                                          │

│   26 │   │   │   setup_logging() # reset since scripts can hijaack logging                                           │

│ in exec_module:883                                                                                                   │

│ in _call_with_frames_removed:241                                                                                     │

│                                                                                                                      │

│ C:\Users\TestBuild\Diffusion\automatic\extensions\sd-webui-model-mixer\scripts\model_mixer.py:34 in   │

│ <module>                                                                                                             │

│                                                                                                                      │

│     33 from copy import copy, deepcopy                                                                               │

│ ❱   34 from modules import script_callbacks, sd_hijack, sd_models, sd_vae, shared, ui_settings,                      │

│     35 from modules import scripts, cache, devices, lowvram, deepbooru, images, paths                                │

╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

ImportError: cannot import name 'ui_settings' from 'modules' (unknown location)

**Comment by wkpark** [comment]
currently, `ui_settings` not used.
~~~diff
diff --git a/scripts/model_mixer.py b/scripts/model_mixer.py
index 8f6b5e0..8caf44b 100644
--- a/scripts/model_mixer.py
+++ b/scripts/model_mixer.py
@@ -31,7 +31,7 @@ import numpy as np
 from PIL import Image

 from copy import copy, deepcopy
-from modules import script_callbacks, sd_hijack, sd_models, sd_vae, shared, ui_settings, ui_common
+from modules import script_callbacks, sd_hijack, sd_models, sd_vae, shared, ui_common
 from modules import scripts, cache, devices, lowvram, deepbooru, images, paths
 from modules import sd_unet
 from modules.generation_parameters_copypaste import parse_generation_parameters
~~~
you can try deleting `ui_settings,` and see what happens.

**Comment by ThePixelDiffusionPirate** [comment]
Thanks, but unfortunately it didn't work. Then came the next error: SD_Unet, I deleted it, then came cache, import deleted, then came a cache error. Unfortunately it doesn't seem to work with SD.Next in this state. Since I have no idea about python, I don't want to mess around with it too much^^

**Reply by wkpark** [reply]
I've just installed the sd.next to see what happen but with no luck.

I will look into SD.Next further, try to fix this issue but,
It seems SD.Next does not sync with the latest sd-webui frequently or has already diverged too much.

**Reply by ThePixelDiffusionPirate** [reply]
Thank you! yes, SD.Next has indeed developed in a somewhat more independent direction recently, that's true. But it really has a lot of features, is constantly being developed further and now has a considerable fan base. Especially for people like me, who are using AMD on a Windows machine, it is actually the only real alternative with DirectML and ONNX/Olive. I just think it's good that they don't just rely on NVIDIA. It's best to take a look at the DEV branch, as it will soon be stable again anyway. 

And Supermerger worked quite well, but I don't need everything and unfortunately it's a bit quiet there at the moment. Probably vacation. But at the moment it's a bit buggy and that's how I came across the model mixer :)

But I have no desire and no money to buy a new graphics card. As long as I can still develop decent games in Unreal Engine 5 with the 5700XT, it's just unnecessary....

**Comment by wkpark** [comment, marked as answer]
https://github.com/wkpark/sd-webui-model-mixer/pull/116 minimal fixes applied to support SD.Next.

**Reply by ThePixelDiffusionPirate** [reply]
cool, i will test it as soon as it is available in the master. Thanks for your effort :-)

**Reply by ThePixelDiffusionPirate** [reply]
i have done several quick n dirty merges without testing everything. for my purposes it seems to work :)
